File: Solutions/1-100/p2.py
"""Solution to problem 2

Again, a simple brute force calculation
"""


def main():
    """main function
    """
    # initiate fibonacci sequence
    fibon_list = [1, 1]

    # generate sequence elements
    small = True
    while small:
        new_element = fibon_list[-1] + fibon_list[-2]
        fibon_list.append(new_element)
        if new_element > 4000000:
            small = False

    # filter out all even elements
    fibon_even = [number for number in fibon_list if number % 2 == 0]

    print(f"Answer: {sum(fibon_even)}")


if __name__ == "__main__":
    main()

File: Solutions/1-100/p2.rs
// Solution to problem 2 in rust

fn main() {
    // create a vector for the Fibonnacci elements (mutable, of course)
    let mut fibon_elements: Vec<u64> = vec![1; 2];
    // create mutable tracker for the while loop
    let mut small = true;
    // needs to be u64 so that we can compare it to the elements in the vector
    let limit: u64 = 4000000;
    while small {
        // sum the last two elements
        let len = fibon_elements.len();
        let new_element = fibon_elements[len - 1] + fibon_elements[len - 2];
        // add the new element to the vector
        fibon_elements.push(new_element);
        // if we are above our limit, abort loop by changing small
        if new_element > limit {
            small = false
        }
    }
    // only keep even element by using summation from iterator
    // need to assign type to the sum
    fibon_elements.retain(|&i|i % 2 == 0);
    let summe: u64 = fibon_elements.iter().sum();
    println!("The answer is: {}", summe)
}
